dali/imgcodec/decoders/nvjpeg2k/nvjpeg2k_test.cc

```diff
@@ -121,14 +134,22 @@ class NvJpeg2000DecoderTest : public NumpyDecoderTestBase<GPUBackend, OutputType
     return opts;
   }
 
-  void RunSingleTest(const ImageTestingData &data) {
+  template<class T, class U>
+  void AssertEqual(const T &img, const U &ref, std::optional<float> eps) {
+    if (eps)
+      this->AssertClose(img, ref, eps.value());
+    else
+      this->AssertEqualSatNorm(img, ref);
+  }
+
+  void RunTest(const ImageTestingData &data, std::optional<float> eps = {}) {
     ImageBuffer image(data.img_path);
     auto decoded = this->Decode(&image.src, this->GetParams(), data.roi);
     auto ref = this->ReadReferenceFrom(data.ref_path);
-    this->AssertEqualSatNorm(decoded, ref);
+    this->AssertEqual(decoded, ref, eps);
   }
 
-  void RunBatchTest(const std::vector<ImageTestingData> &data) {
+  void RunTest(const std::vector<ImageTestingData> &data, std::optional<float> eps = {}) {
```

Comment:
Did you mean 0.0f or std::nullopt for default value of eps?

Reply:
I meant the std::nullopt. Do you suggest to write it there explicitly, or is it fine as it is?

Reply:
Yes, I'd prefer more explicit version `= std::nullopt`